### DISCUSSION TITLE
Instead of requiring us to add the extra CMILib, just compile it with each plugin.

### DISCUSSION THREAD

**Discussion by TomLewis** [Zrips/CMI#6179] [Enhancement]
I understand CMILib is separate so you can share a codebase between your plugins, but there is no need to have this an extra downloadable that could cause extra issues if you not not have the correct version to match the version of the plugin you are using.
Can I suggest you keep CMILib as a tool for you, so when you compile your plugins its just automatically added so its just 1 less thing required by customers/users.

Theres are no other plugins that use it.

**Comment by gecko10000** [comment]
The reason it's currently split is because CMI is pushing the 4 MB Spigot upload limit.

**Comment by Zrips** [comment]
To have one single library means that applying fix to it then it applies same fix to all plugins which uses it. This saves allot of my time and yours and prevents issues with bugs where you could have in base code, like some bug with GUI handling, then i would need to release update for every single plugin and you would need to redownload every single plugin you are using just to fix that. 
There was a reason why it got separated and there is a reason why bigger plugins, like McMMO uses same thing by having McCore.
In short, plugins wont include CMILib in their code, its not efficient and not needed.

**Comment by Baterka** [comment]
For example I am using all Zrips plugins.. It is nice to have just single lib and not need to download 4 times 4MB. So this post seems like nonsense to me.
